.github/workflows/android-unit-tests.yml

```diff
@@ -41,51 +45,51 @@ jobs:
         with:
           distribution: 'temurin'
           java-version: '11'
-      - name: Get yarn cache directory path
+      - name: ♻️ Get yarn cache directory path
         id: yarn-cache-dir-path
         run: echo "::set-output name=dir::$(yarn cache dir)"
-      - uses: actions/cache@v2
-        name: Restore yarn cache
+      - name: ♻️ Restore yarn cache
+        uses: actions/cache@v2
         with:
           path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
           key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
           restore-keys: |
             ${{ runner.os }}-yarn-
-      - run: yarn install --frozen-lockfile
-      - uses: actions/cache@v2
-        name: Restore Gradle cache
+      - name: 🧶 Install node modules in root dir
+        run: yarn install --frozen-lockfile
+      - name: ♻️ Restore node modules in tools
+        uses: actions/cache@v2
+        with:
+          path: tools/node_modules
+          key: ${{ runner.os }}-tools-modules-${{ hashFiles('tools/yarn.lock') }}
+      - name: ♻️ Restore Gradle caches
+        uses: actions/cache@v2
         with:
           path: ~/.gradle/caches
           key: ${{ runner.os }}-gradle-${{ hashFiles('android/*.gradle*') }}
           restore-keys: |
             ${{ runner.os }}-gradle-
-      - uses: actions/cache@v2
-        name: Restore NDK cache
-        id: cache-android-ndk
-        with:
-          path: /usr/local/lib/android/sdk/ndk/21.4.7075529/
-          key: ${{ runner.os }}-ndk-21.4.7075529
-          restore-keys: |
-            ${{ runner.os }}-ndk-
-      - name: Install NDK
-        if: steps.cache-android-ndk.outputs.cache-hit != 'true'
-        run: |
-          sudo $ANDROID_SDK_ROOT/tools/bin/sdkmanager --install "ndk;21.4.7075529"
       - run: echo "$(pwd)/bin" >> $GITHUB_PATH
-      - name: Patch react-native to support single abi
+      - name: 🔧 Patch react-native to support single abi
         run: sed -i 's/^APP_ABI := .*$/APP_ABI := $(if $(NDK_ABI_FILTERS),$(NDK_ABI_FILTERS),$(armeabi-v7a x86 arm64-v8a x86_64))/g' ReactAndroid/src/main/jni/Application.mk
         working-directory: react-native-lab/react-native
-      - name: Run Spotless lint check
-        env:
-          ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/21.4.7075529/
+      - name: 🎸 Run Spotless lint check
         working-directory: android
         run: ./gradlew spotlessCheck || { echo '::error Spotless lint failed. Run `./gradlew spotlessApply` to automatically fix formatting.' && exit 1; }
-      - name: Run native Android unit tests
+      - name: 🎸 Run native Android unit tests
         timeout-minutes: 30
-        env:
-          ANDROID_NDK_HOME: /usr/local/lib/android/sdk/ndk/21.4.7075529/
         run: expotools native-unit-tests --platform android
-      - name: Save test results
+      - name: 🎸 Run instrumented unit tests
```

Comment:
Do you think we need to run instrumented tests in all cases? It takes a lot of time, so maybe we can run them only when the specific packages change?

Reply:
So maybe we can skip it when none of those four packages changed?

Reply:
there's down side from testing specific modules. e.g. the sdk packages ci job occasionally break but pr checks don't alarm from reviewing time. it's because changing expo-modules-core or other core packages may have impact to other modules.  if we can run android test on linux runner and don't wait too much time. i would prefer to test all android tests as it is.

the ci job i really want to test based on changes is the test-suite job. it takes longer time and two macos runners 😞